k8s/upgrade-job/src/events/event_recorder.rs

```diff
@@ -179,41 +179,38 @@ impl EventRecorderBuilder {
                     secondary: event.secondary.clone(),
                 };
                 if let Err(error) = event_handler.publish(event).await.context(EventPublish) {
-                    tracing::error!(?error, "Failed to publish event");
+                    error!(%error);
                 }
 
                 select! {
                     _ = sleep(Duration::from_secs(1200)) => {}
                     event = rx.recv() => { current_event = event }
                 }
             }
-        })
-        .abort_handle();
 
-        Ok(EventRecorder {
-            event_loop_killswitch: aborter,
-            event_sender: tx,
-            from_version,
-            to_version,
-        })
+            // Acknowledgement for exit of event loop, i.e. all events published, nothing left
```

Comment:
Don't think you need this, on the other side you just need to await the task completion?